.github/workflows/build.yml

```diff
@@ -55,3 +55,39 @@ jobs:
 
       - name: static analysis
         run: pio check --skip-packages --fail-on-defect high
+
+  publish-build:
+    needs: build-test
+    if: (github.ref == 'refs/heads/main' && github.event_name == 'push')
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/pip
+            ~/.platformio/.cache
+          key: ${{ runner.os }}-pio
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
```

Comment:
Python 3.11 is pretty old at this point. Maybe is worth start with 3.13 here to save some future headaches :)

Reply:
Good observation. I loosened it to 3.x to make this easier to maintain. (I often prefer to pin specific versions, but in this case I think it's unlikely that a future 3.x version breaks platformio)